swift/schema.py

```diff
@@ -1190,11 +1190,31 @@ class OpaqueTypeArchetypeType(ArchetypeType):
     See https://docs.swift.org/swift-book/LanguageGuide/OpaqueTypes.html."""
     declaration: OpaqueTypeDecl
 
-class OpenedArchetypeType(ArchetypeType):
+class PrimaryArchetypeType(ArchetypeType):
     pass
 
-class PrimaryArchetypeType(ArchetypeType):
+class LocalArchetypeType(ArchetypeType):
+    pass
+
+class OpenedArchetypeType(LocalArchetypeType):
+    pass
+
+class ElementArchetypeType(LocalArchetypeType):
+    pass
+
+class PackArchetypeType(ArchetypeType):
     pass
+
+class PackType(Type):
+    elements: list[Type]
+
+class PackExpansionType(Type):
+    pattern_type: Type
+    count_type: Type
+
+class PackElementType(Type):
+    pack_type: Type
+
```

Comment:
these can also have their tests grouped, right? They probably can go all under `@qltest.test_with("PackType")` (use `PackType` if the definition is there, and `"PackType"` if it is a kind of "forward" definition).